curl_cffi/requests/websockets.py

```diff
@@ -64,14 +66,17 @@ def recv(self) -> Tuple[bytes, int]:
         """
         chunks = []
         flags = 0
-        # TODO use select here
+        sock_fd = ffi.new("long*")
+        lib.curl_easy_getinfo(self.curl._curl, CurlInfo.ACTIVESOCKET, sock_fd)
```

Comment:
It's more consistent to use something like this:

```
sock_fd = self.curl.getinfo(CurlInfo.ACTIVESOCKET)
```

Reply:
```
sock_fd = self.curl.getinfo(CurlInfo.ACTIVESOCKET)
rlist, _, _ = select.select([sock_fd], [], [], 5.0)
```

Adjusted the way ret_cast_option and ret_option are handled in this commit. Is it possible that there are unknown bugs?

https://github.com/lexiforest/curl_cffi/commit/8a1388fdd32e7b642e7a57cf97b711cca17e1447

Reply:
Similar to this issue https://github.com/lexiforest/curl_cffi/issues/407

Reply:
> ```
> sock_fd = self.curl.getinfo(CurlInfo.ACTIVESOCKET)
> rlist, _, _ = select.select([sock_fd], [], [], 5.0)
> ```
> 
> Adjusted the way ret_cast_option and ret_option are handled in this commit. Is it possible that there are unknown bugs?
> 
> [8a1388f](https://github.com/lexiforest/curl_cffi/commit/8a1388fdd32e7b642e7a57cf97b711cca17e1447)

Our supported platforms are very limited, it should be OK for modern architectures.